src/js/components/Doc.js

```diff
@@ -1,21 +1,62 @@
 import React, { Component } from 'react';
-
+import PropTypes from 'prop-types';
+import URLSearchParams from 'url-search-params';
+import { hpe } from 'grommet/themes';
 import { Box, Grommet } from 'grommet';
 
 import Props from './Props';
 
+const THEMES = {
+  grommet: undefined,
+  hpe,
+};
+
 export default class Doc extends Component {
+  constructor(props, context) {
+    super(props, context);
+    this.onTheme = this.onTheme.bind(this);
+    this.state = {};
+    const { router: { route: { location } } } = this.context;
```

Comment:
As we're approaching the release of React 16/Fiber has anyone investigated what context looks like in Fiber? I did some brief research the other night and it sounds like it will be kept and able to store static values (https://twitter.com/sebmarkbage/status/892226820208185344). It may be worth it to get someone from the core team to discuss their long term goal for context. I can pursue this conversation and follow up.